### DISCUSSION TITLE
Failed to setup, check the logs

### DISCUSSION THREAD

**Discussion by stefanovic45** [alandtse/tesla#1058] [Q&A] [answered]
This is a repost from https://github.com/alandtse/tesla/discussions/1057, since that topic was closed but the problem is not completely solved. 

The public key is accessable. But the integration is stuck on Failed to set up: Check the logs.
However, I cannot find any logs for this integration. (i've enabled debug modus).
There are also no logs present in System logs. Where to find the logs?

**Comment by skipishere** [comment]
https://my.home-assistant.io/redirect/integration/?domain=tesla_custom should take you to the correct page and then click `enable debug logging`

When you click again to disable logging you'll get a file downloaded with the contents of the log while debugging was enabled. Be careful there potentially will be some sensitive information that you might want to redact.

**Comment by stefanovic45** [comment, marked as answer]
Ah thanks! That indeed worked... I hat the message 

`412: {"error":"Account xxxx must be registered in the current region https://fleet-api.prd.eu.vn.cloud.tesla.com, please see https://developer.tesla.com/docs/fleet-api/endpoints/partner-endpoints#register"}`

Turned out that my Tesla HTTP Proxy add on is not doing this correctly. I should be done by this add on right?

I fixed it doing the registration manually.:

`CLIENT_ID="xxxx"
CLIENT_SECRET="xxxx"
AUDIENCE="https://fleet-api.prd.eu.vn.cloud.tesla.com"
# Partner authentication token request
curl --request POST \
  --header 'Content-Type: application/x-www-form-urlencoded' \
  --data-urlencode 'grant_type=client_credentials' \
  --data-urlencode "client_id=$CLIENT_ID" \
  --data-urlencode "client_secret=$CLIENT_SECRET" \
  --data-urlencode 'scope=openid vehicle_device_data vehicle_cmds vehicle_charging_cmds' \
  --data-urlencode "audience=$AUDIENCE" \
  'https://auth.tesla.com/oauth2/v3/token'`

End than with using the output from previous command:
`curl --location 'https://fleet-api.prd.eu.vn.cloud.tesla.com/api/1/partner_accounts' \
--header 'Content-Type: application/json' \
--header 'Authorization: Bearer <output previous command> \
--data '{
    "domain":"<YOURDOMAIN>",
    "csr": "-----BEGIN CERTIFICATE REQUEST-----
<YOUR PUBLIC KEY>
-----END CERTIFICATE REQUEST-----
"
  }'`
